services/store.go

```diff
@@ -181,71 +186,46 @@ outer:
 					if !ok {
 						continue
 					}
-					xfer := decodeTransfer(height, info, evt)
-					if xfer == nil {
-						log.Warnf(
-							"No transfer detected for state %#v in transaction %s at height %d",
-							evt.States, info.TxHash.ToHexString(), height,
-						)
-						continue
-					}
-					if failed {
-						if evt.ContractAddress != ongAddr {
+					//check evm ong event log
+					isEvm, eventLog := checkEvmEventLog(evt)
+					if isEvm {
+						xfer, err := parseEvmOngTransferLog(eventLog)
+						if err != nil {
+							log.Warnf("prase evm ong err:%s,height:%d,txhash:%s", err, info.TxHash.ToHexString(), height)
 							continue
 						}
-						// NOTE(tav): We skip additional transfer events on
-						// failed transactions if we've already matched the gas
-						// fee.
-						if gasVerified {
+						balanceCal(xfer, evt, diffs, txn.Transfers)
```

Comment:
Should return `transfers` within `balanceCal` and assign it back to `txn.Transfers`.

Reply:
like this:           txn.Transfers = append(txn.Transfers, balanceCal(xfer, evt, diffs))